datastore/googlepubsub/publisher.go

```diff
@@ -81,14 +97,12 @@ func (p *Producer) Produce(entry *telemetry.Record) {
 	})
 	if _, err = result.Get(ctx); err != nil {
 		p.logger.Errorf("pubsub_err err: %v", err)
-		if p.prometheusEnabled {
-			metrics.StatsIncrement(p.statsCollector, "pubsub_publish_total", 1, map[string]string{"record_type": entry.TxType})
-			metrics.StatsIncrement(p.statsCollector, "pubsub_publish_total_bytes", int64(entry.Length()), map[string]string{"record_type": entry.TxType})
-		} else {
-			metrics.StatsIncrement(p.statsCollector, entry.TxType+"_produce", 1, map[string]string{})
-		}
+		metricsRegistry.publishCount.Inc(map[string]string{})
+
+		metricsRegistry.publishCount.Inc(map[string]string{"record_type": entry.TxType})
+		metricsRegistry.publishBytesTotal.Add(int64(entry.Length()), map[string]string{"record_type": entry.TxType})
 	} else {
-		metrics.StatsIncrement(p.statsCollector, "pubsub_err", 1, map[string]string{})
+		metricsRegistry.errorCount.Inc(map[string]string{})
```

Comment:
For error count, is there any reason to not report the "record_type"? @aaronpkahn

Reply:
yeah we should add the record type